packages/owlbot-bootstrapper/common-container/split-repo.ts

```diff
@@ -0,0 +1,181 @@
+ * @param githubToken a short-lived access Github access token
+ */
+export class SplitRepo {
+  language: Language;
+  repoName: string;
+  apiId: string;
+  octokit: Octokit;
+  githubToken?: string;
+
+  constructor(
+    language: Language,
+    apiId: string,
+    octokit: Octokit,
+    githubToken?: string
+  ) {
+    this.language = language;
+    this.apiId = apiId;
+    this.repoName = this._createRepoName(this.language, this.apiId);
+    this.githubToken = githubToken;
+    this.octokit = octokit;
+  }
+  /**
+   * Creates a new repo in github
+   *
+   * @param octokit the url for the repo to clone without the preceeding https://, e.g., just github.com/googleapis/nodejs-kms.git
+   * @param repoName the name of the repo to create, i.e., python-kms
+   */
+  public async _createRepo(octokit: Octokit, repoName: string) {
+    try {
+      await octokit.rest.repos.createInOrg({
+        org: ORG,
+        name: repoName,
+      });
+    } catch (err) {
+      if ((err as any).message.match(/name already exists on this account/)) {
+        logger.info(
+          `${ORG}/${repoName} already exists, skipping repo creation`
+        );
+      } else {
+        logger.error(err as any);
+        throw err;
+      }
+    }
+  }
+
+  /**
+   * Creates a repo name from the api Id
+   *
+   * @param language the Language that is triggering the process
+   * @param apiId the api ID that is triggering the process, i.e., google.cloud.kms.v1
+   */
+  public _createRepoName(language: string, apiId: string): string {
+    const apiIdSplit = apiId.split('.');
+    if (apiIdSplit[1] === 'cloud') {
+      return `${language}-${apiIdSplit[2]}`;
+    } else {
+      return `${language}-${apiIdSplit[1]}-${apiIdSplit[2]}`;
+    }
+  }
+
+  /**
+   * Initializes an empty git repo locally
+   *
+   * @param repoName the name of the git repo to initialize
+   * @param directoryPath the path where the empty repo should be initialized
+   */
+  public async _initializeEmptyGitRepo(
+    repoName: string,
+    directoryPath: string
+  ) {
+    cmd(`mkdir ${repoName}`, {cwd: directoryPath});
+    cmd('git init', {cwd: `${directoryPath}/${repoName}`});
+  }
+
+  /**
+   * Commits any changes made to main and pushes them
+   *
+   * @param repoName the name of the repo on github to push the changes to
+   * @param directoryPath name of the directory in which the process is running (i.e., 'workspace' for a container)
+   * @param githubToken a short-lived access Github access token
+   * @param repoUrl the url of the repo to push to; mostly for testing purposes.
+   */
+  public async _commitAndPushToMain(
+    repoName: string,
+    directoryPath: string,
+    githubToken?: string,
+    repoUrl?: string
+  ) {
+    if (githubToken) {
+      cmd(
+        `git add .; git commit -m "feat: adding initial files"; git branch -M main; git remote add origin https://x-access-token:${githubToken}@github.com/${ORG}/${repoName}; git push -u origin main`,
```

Comment:
Since this is shelling out to call `git`, we should ensure that our container has the necessary packages installed